app/assets/javascripts/discourse/components/badge-title.js.es6

```diff
@@ -1,12 +1,30 @@
 import { ajax } from "discourse/lib/ajax";
-import BadgeSelectController from "discourse/mixins/badge-select-controller";
+import computed from "ember-addons/ember-computed-decorators";
 
-export default Ember.Component.extend(BadgeSelectController, {
+export default Ember.Component.extend({
   classNames: ["badge-title"],
 
   saved: false,
   saving: false,
 
+  @computed("saving")
+  savingStatus(saving) {
+    return saving ? I18n.t("saving") : I18n.t("save");
+  },
+
+  @computed("selectableUserBadges", "selectedUserBadgeId")
+  selectedUserBadge(selectableUserBadges, selectedUserBadgeId) {
+    selectedUserBadgeId = parseInt(selectedUserBadgeId);
+    let selectedUserBadge = null;
+
+    selectableUserBadges.forEach(function(userBadge) {
```

Comment:
Should this keep looping after it's already found the id?

It seems to me we could use `return selectableUserBadges.findBy('id', selectedUserBadgeId)`